Automattic/woocommerce-payments
# Place WCPay subscriptions on-hold prior to renewal

Fixes #3923

#### Changes proposed in this Pull Request

Temporarily place the subscription on-hold to imitate the normal subscription renewal flow. This ensures the downstream effects take place, e.g. a payment status order note is added and the `woocommerce_subscription_payment_complete` action is fired.

#### Testing instructions

1. Create a small plugin that creates a log entry when the `woocommerce_subscription_renewal_payment_complete` action runs. (see below for example)
2. As a shopper, purchase a subscription product.
3. Use the WCPay Dev Tools test clock feature to force a renewal.
4. Observe, the log entry is created (this log entry **is not** created on `develop`).

```php
add_action(
	'woocommerce_subscription_renewal_payment_complete',
	function( $subscription ) {
		error_log( sprintf( 'Subscription #%d renewal payment complete!', $subscription->get_id() ) );
	}
);
```

-------------------


**Post merge**
